corehq/motech/repeaters/signals.py

```diff
@@ -100,6 +104,19 @@ def fire_synchronous_case_repeaters(sender, case, **kwargs):
     _create_repeat_records(DataRegistryCaseUpdateRepeater, case, fire_synchronously=True)
 
 
+def create_data_source_transaction_log_entry(sender, **kwargs):
+    """Creates a transaction log for the datasource that changed"""
+    transaction_log = DataSourceRowTransactionLog.objects.create(
+        domain=kwargs["domain"],
+        data_source_id=kwargs["data_source_id"],
+        row_id=kwargs["doc_id"],
+        row_data=kwargs["row_change"],
+        action=kwargs["action"]
+    )
+    create_repeat_records(DataSourceRepeater, transaction_log)
+
+
 successful_form_received.connect(create_form_repeat_records)
 successful_form_received.connect(create_short_form_repeat_records)
 sql_case_post_save.connect(create_case_repeat_records, CommCareCase)
+ucr_data_source_updated.connect(create_data_source_transaction_log_entry)
```

Comment:
Why is this needed here? I thought we were making repeat records when updating data source?

Reply:
Yeah this is where we do it. We call this signal, which then creates the transaction log, and then calls the `create_repeat_record` method to create the record

Reply:
and this when we save rows?
https://github.com/dimagi/commcare-hq/pull/33781/files#diff-1d71ce3e4eed9626ae52cd181a05d9a240094f731406d696625757de1d4efd97R147
Is that for something else?

Reply:
I'm not following. Mind rephrasing the question please?

Reply:
Hey @SmittieC

My bad.
Confirming that this is just linking signal `ucr_data_source_updated` to the function `create_data_source_transaction_log_entry` and not actually calling it.
I misunderstood this as a call to the function `create_data_source_transaction_log_entry` which would create a log entry.

Reply:
All good :+1: